### DISCUSSION TITLE
[Hardware] Micro SD support?

### DISCUSSION THREAD

**Discussion by hahalolha** [joeycastillo/Sensor-Watch#100] [Ideas]
Might be impossible due to case interference however, would be worth investigation. One method to incorporate MicroSD could be to use the rear of the PCB. SD card slot does not need to be connected to SoC, as to act as a 'Cold' slot.

Motivation is: Storage of files (No connection to SoC) or Data logging (Connected to SoC)

**Comment by aethrys** [comment]
For your "cold storage" idea, what you've described is how I currently use my watch...

I use NODE's Data Runner extension (1) that slightly increases the height of the watch, but gives it enough room to fit a "push-push" micro SD card socket. This particular part has been discussed here on the project before. You may either purchase directly from NODE (2) or build it yourself (3) (both will run-you-over $10 depending on where you source parts, and shipping... unless you already have the metal SD card sockets / 3D printer).

Either way, it's been a pretty good addition. I have the watch with me daily, either on my wrist or in a bag. I'm going through a "minimalist" phase in my life, and shrinking down what I carry regularly... and then instead making sure all the "big stuff", like a home server, is always available (via a Wireguard VPN) and organized at my "home base".

On the SD card, I essentially have my whole life on it. There's a partition for Live CDs (ISOs) made bootable using Ventoy, a partition for a portable terminal / shell that includes binaries for both Linux and Windows, and then finally another partition for persistent files like my deployment scripts, alias files, dot files, etc (managed and replicated to the SD card via GIT, and also encrypted). I would "more literally" have my whole life on it if i included my personal notes and journal, but those are stored on a separate encrypted SD card I have tucked away.

My point here is there's a lot you can do right now while you wait for your board to arrive. Going into this and setting it all up, I had the intention of using the One-Time-Password feature of the Sensor Watch, to create a "two-step mobile compute environment" opposed to a "one-step" like with a rooted Android smartwatch loaded up with Termux or something. 

You could take it further... I believe you can fit two SD cards inside the Data Runner extension, one in the push-push socket, and another inside the watch under the metal backplate. I haven't tested the tolerances for this, though.

The Sensor Watch's PCB is pretty cramped currently, but would be cool if it did support "hot" features of mounting the card to the ARM chip. Alternatively, if instead the only room allowed is outside the absolute interior of the watch, like with the Data Runner, there's still more to be done. The "cold" SD storage could be made more accessible by including an additional PCB or SD to USB adapter. There are micro SD to USB A adapters that exist which are completely flat, but friction fit. One could take apart or mimic one of those, and then extend the USB A traces out to the exterior of watch. At that point you can make the female connection to this whole "Frankenstein SD card adapter" whatever you want... but it would still be on your wrist, as opposed to having to take the SD card out every time you want to use it... Maybe include a magnetic USB tip (4) so the port doesn't get worn out.

If you went through all that effort, it wouldn't be much work further to include an always-attached retractable cable (5)... then you _definitely_ could pretend you're a cyborg living in the future (*cough* Cyberpunk *cough*).

Cheers.


[1 - Datarunner](https://n-o-d-e.net/datarunner.html)
[2 - NODE Store](https://n-o-d-e.shop/collections/frontpage/products/f-91w-micro-sd-addon?variant=39419110686818)
[3 - Datarunner project files](https://n-o-d-e.live/files/data_runner/data_runner.zip)
[4 - Retractable USB cable example](https://www.amazon.com/Cable-Matters-Gold-Plated-Retractable-Micro-USB/dp/B00SVVY844)
[5 - Magnetic USB Tip example](https://www.amazon.com/s?k=magnetic+usb+tip&crid=GWBH8LDOO2SZ&sprefix=magnetic+usb+tip%2Caps%2C112&ref=nb_sb_noss_1)
